[13.0][MIG] partner_multi_company

JoanSForgeFlow opened the pull request:
**partner_multi_company** migration to v13.
Depends on:
- #197

cc ~ @LoisRForgeFlow @JordiBForgeFlow

oihane committed:
New module partner_multi_company

pedrobaeza committed:
partner_multi_company: Tests + README

pedrobaeza committed:
partner_multi_company: Sync company_ids field

pedrobaeza committed:
product_multi_company: Avoid errors in other tests

This fix prevents errors when there are other tests that create partners
and base_suspend_security is still not ready. There can be still problems
if these partners are not created with SUPERUSER.

pedrobaeza committed:
partner_multi_company: Migration to 9.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

lasley committed:
[MIG] partner_multi_company: Upgrade to v10
* Bump versions
* Rename manifest
* Rename openerp to odoo
* Implement base_multi_company

Add depends

Swap dependencies

Add base_suspend_security depend back & server-tools repo

Fix tests

Add test for partner._commercial_fields

Why not both?!

pedrobaeza committed:
[MIG] *_multi_company: OpenUpgrade migration scripts

oca-transbot committed:
OCA Transbot updated translations from Transifex

cubells committed:
[MIG] partner_multi_company: Migration to 11.0

pedrobaeza committed:
[MIG] partner_multi_company: Finish migration to 11.0

oca-travis committed:
[UPD] Update partner_multi_company.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-11.0/multi-company-11.0-partner_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-11-0/multi-company-11-0-partner_multi_company/

kittiu committed:
[12.0][MIG] partner_multi_company

oca-travis committed:
[UPD] Update partner_multi_company.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-12.0/multi-company-12.0-partner_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-partner_multi_company/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-12.0/multi-company-12.0-partner_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-partner_multi_company/

MiquelRForgeFlow committed:
[FIX] If user switch company, don't delete user's partner's other companies

OCA-git-bot committed:
partner_multi_company 12.0.1.0.1

JoanSForgeFlow committed:
[IMP] partner_multi_company: black, isort

JoanSForgeFlow committed:
[MIG] partner_multi_company: Migration to 13.0

JordiBForgeFlow commented:
@JoanSForgeFlow please add the oca_dependencies.txt file and include the reference to the base_multi_company, like it's done here 068c50972ca74dba0eeb148e68a4e40f5baefdc2

This will make it easier for reviewers to test using runbot. Now everything fails due to the missing dependency.

JoanSForgeFlow committed:
added migration scripts

AaronHForgeFlow committed:
[UPT]partner_multi_company/migrations do not keep old table + update rule

AaronHForgeFlow commented:
Do not merge yet. Last commit will break tests for account_invoice_multi_company when partner_multi_company is not installed. The only solution I see is separate in the CI the installation of this module from account_invoice_inter_company and, sale_purchase_inter_company

pedrobaeza commented:
If there's an incompatibility between 2 modules, better to use `config['test_enable']` and a context for detecting the testing of the module for enabling/disabling the incompatible behavior. Example:

https://github.com/OCA/bank-payment/blob/3c96fedbb95060c9cd3a2aa87d383324d486bee1/account_payment_transfer_reconcile_batch/models/payment_order.py#L28

AaronHForgeFlow commented:
I could not do that, because the error will raise in a constraint in standard Odoo when loading a file specified in the test. I tried to overwrite the records created in that file in the partner_multi_company module. But that was too late, because the original file will always be leaded as it is. The only way I found if to include a separate file and load one file or the other depending if the field company_ids exists or not in the partner.

pedrobaeza commented:
Well, unless you show me through Travis log, I can't tell you more.

pedrobaeza commented:
What I can say is that the XML import done in `account_invoice_inter_company` is very ugly and should be changed by conventional `setUp`.

AaronHForgeFlow commented:
Basically, the file creates the partner without a company as company partner. But this https://github.com/OCA/multi-company/blob/13.0/base_multi_company/models/multi_company_abstract.py#L24 assign a a value in company_ids, and consequently to the company_id field. Then an error will raise because the company of the partner does not match the company of the warehouse of the company.

I agree that loading that file that way is ugly, but it was done like that for a reason: https://github.com/OCA/multi-company/commit/e4ffc42837a7d82e8d8ab50bf883f82dc89e0ef1 and that reason seems still valid.

LoisRForgeFlow commented:
Another option is to add a travis task excluding that module.

pedrobaeza commented:
I'm not talking about returning to demo data, but create them in the `setUp`/`setUpClass` method. Anyway, I don't see this red. Or do you need to rebase? Please show me a red Travis with the error for being able to tell you how to fix it without isolating modules, which is a pain.

AaronHForgeFlow commented:
It is not red because I am loading different files depending if the module partner_multi_company is installed or not. Modules are not isolated and work together perfectly, the problem is the data loaded. I agree better solution is to create those records in the SetUpClass.

pedrobaeza commented:
Are you doing it or do you want me to do it?

AaronHForgeFlow commented:
And that is not the only module that loads data this way. In the end all those modules need a change in the way they load data for tests.

AaronHForgeFlow commented:
> Are you doing it or do you want me to do it?

I am doing that.

pedrobaeza commented:
OK, thanks. Contact me by chat if you want that both see anything in live.

AaronHForgeFlow commented on account_invoice_inter_company/tests/test_inter_company_invoice.py:
```diff
@@ -14,44 +14,591 @@ class TestAccountInvoiceInterCompanyBase(SavepointCase):
+                "company_id": cls.company_a.id,
+            }
+        )
+        cls.invoice_company_a = cls.env["account.move"].create(
```
Seems this line is failing:

`
 Remaining fields to compute before setting up registry: defaultdict(<class 'set'>, {account.move.amount_tax_signed: {<NewId 0x7f33b94a96d8>}, account.move.amount_total_signed: {<NewId 0x7f33b94a96d8>}, account.move.amount_residual: {<NewId 0x7f33b94a96d8>}, account.move.invoice_payment_state: {<NewId 0x7f33b94a96d8>}, account.move.amount_total: {<NewId 0x7f33b94a96d8>}, account.move.amount_residual_signed: {<NewId 0x7f33b94a96d8>}, account.move.amount_untaxed_signed: {<NewId 0x7f33b94a96d8>}, account.move.amount_untaxed: {<NewId 0x7f33b94a96d8>}, account.move.amount_tax: {<NewId 0x7f33b94a96d8>}, account.move.line.account_root_id: {<NewId 0x7f33baf29978>}, account.move.line.account_internal_type: {<NewId 0x7f33baf29978>}, account.move.line.amount_residual_currency: {<NewId 0x7f33baf29978>}, account.move.line.reconciled: {<NewId 0x7f33baf29978>}, account.move.line.amount_residual: {<NewId 0x7f33baf29978>}, account.move.line.tax_audit: {<NewId 0x7f33baf29978>}})
`

AaronHForgeFlow commented:
Do not merge yet. Still need to convert tests for purchase_sale_inter_company.

AaronHForgeFlow reviewed (approved):
Test refactored completed. This is now ready for review :)

LoisRForgeFlow commented:
Just fixed conflicts.

LoisRForgeFlow commented:
This module needs https://github.com/odoo/odoo/pull/59237 in order to not raise an error when trying to reconcile an statement.

juliap98 reviewed (approved):
functional review, LGTM

LoisRForgeFlow committed:
[13.0][FIX] base_multi_company: name_search

In some situations the 'in' operator is used with company_id in a
name_search. ORM does not convert to a proper WHERE clause when using
the 'in' operator.
e.g: ```
     WHERE "res_partner"."id" in (SELECT "res_partner_id"
     FROM "res_company_res_partner_rel" WHERE "res_company_id" IN (False, 1)
```
patching the args to expand the cumbersome args int a OR clause fix
the issue.
e.g: ```
     WHERE "res_partner"."id" not in (SELECT "res_partner_id"
             FROM "res_company_res_partner_rel"
             where "res_partner_id" is not null)
         OR  ("res_partner"."id" in (SELECT "res_partner_id"
             FROM "res_company_res_partner_rel" WHERE "res_company_id" IN 1)
```

pedrobaeza commented:
Rebased on #266 for seeing if Travis is OK.
